espresso/core/java/androidx/test/espresso/matcher/ViewMatchers.java

```diff
@@ -2225,4 +2229,13 @@ private static String safeGetResourceEntryName(Resources res, int id) {
   private static boolean isViewIdGenerated(int id) {
     return (id & 0xFF000000) == 0 && (id & 0x00FFFFFF) != 0;
   }
+
+  @Nullable
+  private static String highlightNbsp(@Nullable CharSequence text) {
+   if (text == null) {
+       return null;
+   }
+   return text.toString().replace(NBSP_SYMBOL, "[NBSP]");
```

Comment:
This only helps with nbsp. What about other invisible characters? e.g. "\n" vs "\r\n".
Or any of the other ones supported in IDEA?
https://github.com/JetBrains/intellij-community/commit/d210ff3bd19479ab37862f1caa186d5b4fd73688#diff-0038f42400cf76ea0b815fd056c81c2c84d6bb4acf27ed3cbc7e4986024c7cddL31-L45

Reply:
Since we tend to use "\n" only, I didn't face any confusion with \r\n".
For now, nbsp is a sore, and I don't think we should treat all of them at once.

Reply:
Oh the link is borked, I meant to link here:
https://github.com/JetBrains/intellij-community/blob/master/platform/platform-impl/src/com/intellij/openapi/editor/impl/view/SpecialCharacterFragment.java#L19-L44
There are [many many different whitespace characters](https://jkorpela.fi/chars/spaces.html) that will result in the same experience as nbsp. \r\n was just an example.

Reply:
Wow... Well, this way we either support nbsp only (unless it's sufficient) or add a handful of all characters. In this case, we should fall back to `[nbsp]` notation because we don't have enough delegates like `␣`.
I still stick to this single case idea, if you ask me. What do you think?

Reply:
I'll defer to @brettchabot or maintainers here, I personally would go all in, since it gives better developer experience, and the list can be based on IJ; after all you wanted "the way AS handles this" 😊.

Reply:
I agree with @TWiStErRob: [nbsp] is an example of special characters. We don't have to handle all of them at once, but I think we probably want to keep the implementation open to adding more special characters in the future.

Suggest putting the logic of special character handling in a util class, probably under https://github.com/android/android-test/tree/master/espresso/core/java/androidx/test/espresso/util. It can be used by ViewMatchers and other classes too.

Reply:
Ok, I did it! :)
Moved logic with characters to a separate class. 
Added test cases.